Description: Fixed crash which sometimes happens on shutdown of the CUPS
 daemon, caused by a wrong shutdown sequence for shutting down the Avahi
 threaded poll.
Author: Till Kamppeter <till.kamppeter@gmail.com>
Bug-Upstream: http://cups.org/str.php?L4180
Bug-Upstream: http://cups.org/str.php?L4213
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1034045
Last-Updated: 2012-10-16

--- a/scheduler/dirsvc.c
+++ b/scheduler/dirsvc.c
@@ -1333,6 +1333,7 @@
   DNSSDMaster = NULL;
 
 #  else /* HAVE_AVAHI */
+  avahi_threaded_poll_stop(DNSSDMaster);
   avahi_client_free(DNSSDClient);
   DNSSDClient = NULL;
 
